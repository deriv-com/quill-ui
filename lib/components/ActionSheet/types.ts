import { ComponentPropsWithoutRef } from "react";
import { ExcludeAllNull, TLeftOrRight } from "../../types";

interface OpenType {
    isOpen?: boolean | undefined;
    onOpen?: () => void;
}

interface actionSheetRootCVA {
    position?: TLeftOrRight;
    show?: boolean;
}

interface actionSheetFooterCVA {
    alignment?: "vertical" | "horizontal";
}

export interface RootProps
    extends ComponentPropsWithoutRef<"div">,
        ExcludeAllNull<actionSheetRootCVA>,
        OpenType {
    onClose?: () => void;
    type?: "modal" | "non-modal";
    expandable?: boolean;
}

export type RootPosition = RootProps["position"];

interface ActionType {
    content: string;
    onAction: () => void;
}

export interface FooterProps
    extends ComponentPropsWithoutRef<"div">,
        actionSheetFooterCVA {
    primaryAction?: ActionType;
    secondaryAction?: ActionType;
    shouldCloseOnPrimaryButtonClick?: boolean;
    shouldCloseOnSecondaryButtonClick?: boolean;
}

export type FooterAlignment = FooterProps["alignment"];
