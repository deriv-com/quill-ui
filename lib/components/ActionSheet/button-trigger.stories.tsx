import type { Meta, StoryObj } from "@storybook/react";
import { fn } from "@storybook/test";
import { ActionSheetExample } from "./mocks/example";
import {
    LabelPairedPlaceholderCaptionBoldIcon,
    StandaloneXmarkRegularIcon,
} from "@deriv/quill-icons";

const meta: Meta = {
    title: "Components/Action Sheet/Button Trigger",
    component: ActionSheetExample,
    tags: ["autodocs"],
    args: {
        onClose: fn(),
        title: "Title",
        description: "Description",
        closeIcon: <StandaloneXmarkRegularIcon />,
        icon: <LabelPairedPlaceholderCaptionBoldIcon />,
    },
    argTypes: {
        show: { table: { disable: true } },
        handleOpen: { table: { disable: true } },
        handleClose: { table: { disable: true } },
        isOpen: {
            description:
                "If you wish to manage the opening and closing states, transmit the `open` state from your component. Set the initial value to `false` when passing it.",
        },
        onOpen: {
            description:
                "Pass your callback function using this method. It will be triggered on the open function. If you are passing the `isOpen` state, provide your `open` `setState` function like this: `onOpen={() => setIsOpen(true)}`.",
        },
        onClose: {
            description:
                "Pass your callback function using this method. It will be triggered on the open function.",
        },
        expandable: {
            control: { type: "boolean" },
            description:
                "This prop controls the expandability of the bottom sheet.",
        },
        type: {
            options: ["modal", "non-modal"],
            control: { type: "radio" },
            description:
                "This property is used to specify the type, which offers two options: `modal` and `non-modal`. When set to `modal`, it adds an overlay over the entire body, and clicking on it will close the action sheet. On the other hand, when set to `non-modal`, the area outside of the action sheet remains interactive and no overlay.",
        },
        position: {
            options: ["left", "right"],
            control: { type: "radio" },
            description: "This prop will make bottom sheet expandable",
        },
        title: {
            description: "Title for `ActionSheet.Header`",
        },
        description: {
            description: "Description for `ActionSheet.Header`",
        },
        closeIcon: {
            description:
                "This props allowed you to pass in icon for close button in `ActionSheet.Header`",
        },
        icon: {
            description:
                "This props allowed you to pass in icon for `ActionSheet.Header`",
        },
        primaryAction: {
            control: {
                type: "object",
            },
            description:
                "This prop is meant for `ActionSheet.Footer`. It accepts two property: `Content`, which accepts a string, and `onAction`, which takes a function.",
        },
        secondaryAction: {
            control: {
                type: "object",
            },
            description: "Same as `primaryAction`",
        },
        alignment: {
            control: {
                type: "radio",
            },
            options: ["vertical", "horizontal"],
            description:
                "This prop is for `ActionSheet.Footer` buttons alignment",
        },
    },
} satisfies Meta<typeof ActionSheetExample>;

export default meta;
type Story = StoryObj<typeof ActionSheetExample>;

export const ButtonTrigger: Story = {
    args: {
        expandable: true,
        type: "modal",
        position: "right",
        primaryAction: {
            content: "Primary",
            onAction: () => null,
        },
        secondaryAction: {
            content: "Secondary",
            onAction: () => null,
        },
        alignment: "vertical",
    },
};
