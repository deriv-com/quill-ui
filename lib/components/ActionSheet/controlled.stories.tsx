import type { Meta, StoryObj } from "@storybook/react";

import { ActionSheetExampleControlled } from "./mocks/example";
import { fn } from "@storybook/test";
import {
    LabelPairedPlaceholderCaptionBoldIcon,
    StandaloneXmarkRegularIcon,
} from "@deriv/quill-icons";

const icons: Record<string, object | null> = {
    with_icon: <LabelPairedPlaceholderCaptionBoldIcon />,
    none: null,
};

const meta: Meta = {
    title: "Components/Action Sheet/Controlled",
    component: ActionSheetExampleControlled,
    tags: ["autodocs"],
    parameters: {
        controls: { hideNoControlsWarning: true },
    },
    args: {
        isOpen: false,
        onClose: fn(),
        title: "Title",
        description: "Description",
        type: "non-modal",
        closeIcon: <StandaloneXmarkRegularIcon />,
        onOpen: fn(),
        isPrimaryButtonDisabled: false,
        isSecondaryButtonDisabled: false,
        shouldCloseOnDrag: true,
        shouldCloseOnPrimaryButtonClick: true,
        shouldCloseOnSecondaryButtonClick: true,
        fullHeightOnOpen: false,
    },
    argTypes: {
        isOpen: {
            description:
                "If you wish to manage the opening and closing states, transmit the `open` state from your component. Set the initial value to `false` when passing it.",
            control: { type: "boolean" },
        },
        isPrimaryButtonDisabled: {
            control: { type: "boolean" },
            description:
                "This prop controls if primary button is disabled or not.",
        },
        isSecondaryButtonDisabled: {
            control: { type: "boolean" },
            description:
                "This prop controls if secondary button is disabled or not.",
        },
        show: { table: { disable: true } },
        shouldCloseOnDrag: {
            control: { type: "boolean" },
            description:
                "This prop controls if action sheet should be closed on drag down or not. Default value: false. Property should be passed to ActionSheet.Portal",
        },
        shouldDetectSwipingOnContainer: {
            table: { type: { summary: "boolean | undefined" } },
            options: ["true", "false"],
            control: { type: "boolean" },
            description:
                "This prop controls if swiping will be detected on the whole Action Sheet, not only on Handlebar. Default value: false. Property should be passed to ActionSheet.Portal",
        },
        showHandlebar: {
            table: { type: { summary: "boolean | undefined" } },
            options: ["true", "false"],
            control: { type: "boolean" },
            description:
                "This prop controls if Handlebar should be visible or not. Default value: true",
        },
        fullHeightOnOpen: {
            description:
                "This props controls is full height when open the actionsheet",
            control: { type: "boolean" },
            table: { type: { summary: "boolean" } },
        },
        handleOpen: { table: { disable: true } },
        handleClose: { table: { disable: true } },
        onOpen: {
            description:
                "Pass your callback function using this method. It will be triggered on the open function.",
        },
        onClose: {
            description:
                "Pass your callback function using this method. It will be triggered on the open function.",
        },
        expandable: {
            control: { type: "boolean" },
            description:
                "This prop controls the expandability of the bottom sheet.",
        },
        type: {
            options: ["modal", "non-modal"],
            control: { type: "radio" },
            description:
                "This property is used to specify the type, which offers two options: `modal` and `non-modal`. When set to `modal`, it adds an overlay over the entire body, and clicking on it will close the action sheet. On the other hand, when set to `non-modal`, the area outside of the action sheet remains interactive and no overlay.",
        },
        position: {
            options: ["left", "right"],
            control: { type: "radio" },
            description: "This prop will make bottom sheet expandable",
        },
        icon: {
            description:
                "This props allowed you to pass in icon for `ActionSheet.Header`",
            options: Object.keys(icons),
            mapping: icons,
            control: {
                type: "radio",
            },
        },
        primaryAction: {
            control: {
                type: "object",
            },
            description:
                "This prop is meant for `ActionSheet.Footer`. It accepts two property: `Content`, which accepts a string, and `onAction`, which takes a function.",
        },
        secondaryAction: {
            control: {
                type: "object",
            },
            description: "Same as `primaryAction`",
        },
        shouldCloseOnPrimaryButtonClick: {
            table: { type: { summary: "boolean | undefined" } },
            options: ["true", "false"],
            control: { type: "boolean" },
            description:
                "This prop controls if Action Sheet should be closed or not when primary button was clicked. Default value: true",
        },
        shouldCloseOnSecondaryButtonClick: {
            table: { type: { summary: "boolean | undefined" } },
            options: ["true", "false"],
            control: { type: "boolean" },
            description:
                "This prop controls if Action Sheet should be closed or not when secondary button was clicked. Default value: true",
        },
        alignment: {
            control: {
                type: "radio",
            },
            options: ["vertical", "horizontal"],
            description:
                "This prop is for `ActionSheet.Footer` buttons alignment",
        },
    },
} satisfies Meta<typeof ActionSheetExampleControlled>;

export default meta;
type Story = StoryObj<typeof ActionSheetExampleControlled>;

export const Controlled: Story = {
    args: {
        expandable: true,
        position: "right",
        primaryAction: {
            content: "Primary",
            onAction: () => null,
        },
        secondaryAction: {
            content: "Secondary",
            onAction: () => null,
        },
        alignment: "vertical",
    },
};
