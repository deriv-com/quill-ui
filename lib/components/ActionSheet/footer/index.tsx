import { FooterProps } from "../types";
import { useContext } from "react";
import { ActionSheetContext } from "../root";
import clsx from "clsx";
import "./footer.scss";
import { Button } from "@components/Button";

const Footer = ({
    primaryAction,
    secondaryAction,
    alignment,
    className,
    shouldCloseOnPrimaryButtonClick = true,
    shouldCloseOnSecondaryButtonClick = true,
    isPrimaryButtonDisabled,
    isSecondaryButtonDisabled,
    ...rest
}: FooterProps) => {
    const { handleClose } = useContext(ActionSheetContext);
    if (!primaryAction && !secondaryAction) return null;

    const primaryActionHandler = () => {
        primaryAction?.onAction();
        if (shouldCloseOnPrimaryButtonClick) handleClose?.();
    };

    const secondaryActionHandler = () => {
        secondaryAction?.onAction();
        if (shouldCloseOnSecondaryButtonClick) handleClose?.();
    };

    return (
        <div
            className={clsx(
                "quill-action-sheet--footer",
                `quill-action-sheet--footer__variant--${alignment}`,
                className,
            )}
            {...rest}
        >
            {primaryAction && (
                <Button
                    onClick={primaryActionHandler}
                    color="black-white"
                    size="lg"
                    label={primaryAction.content}
                    fullWidth
                    disabled={isPrimaryButtonDisabled}
                />
            )}
            {secondaryAction && (
                <Button
                    onClick={secondaryActionHandler}
                    variant="secondary"
                    color="black-white"
                    size="lg"
                    label={secondaryAction.content}
                    fullWidth
                    disabled={isSecondaryButtonDisabled}
                />
            )}
        </div>
    );
};

export default Footer;
