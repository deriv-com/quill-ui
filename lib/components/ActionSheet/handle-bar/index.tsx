import { ComponentProps } from "react";
import "./handle-bar.scss";
import clsx from "clsx";

export interface BarProps extends ComponentProps<"div"> {
    position?: "sticky" | "absolute";
}

const HandleBar = ({ position = "sticky", ...rest }: BarProps) => {
    return (
        <div
            className={clsx(
                "quill-action-sheet--handle-bar",
                `quill-action-sheet--handle-bar--${position}`,
            )}
            data-testid="dt-actionsheet-handle-bar"
            {...rest}
        >
            <span className="quill-action-sheet--handle-bar--line" />
        </div>
    );
};

export default HandleBar;
