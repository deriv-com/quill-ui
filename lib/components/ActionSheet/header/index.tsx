import { ComponentPropsWithoutRef, ReactNode, useContext } from "react";
import { ActionSheetContext } from "../root";
import clsx from "clsx";
import "./header.scss";
import { Heading, Text } from "../../Typography";

interface HeaderProps extends ComponentPropsWithoutRef<"div"> {
    title?: string;
    description?: string;
    icon?: ReactNode;
    closeIcon?: ReactNode;
}

const Header = ({
    className,
    title,
    description,
    icon: Icon,
    closeIcon: CloseIcon,
    ...rest
}: HeaderProps) => {
    const { expandable } = useContext(ActionSheetContext);
    const { handleClose } = useContext(ActionSheetContext);

    return (
        <div
            className={clsx(
                "quill-action-sheet--header",
                `quill-action-sheet--header__expandable--${expandable}`,
                className,
            )}
            data-testid="action-sheet-header"
            {...rest}
        >
            <div className="quill-action-sheet--title">
                <div className="quill-action-sheet--title--icon">{Icon}</div>
                <Heading.H5>{title}</Heading.H5>
                <button
                    className="quill-action-sheet--title--icon quill-action-sheet--title--icon--close"
                    onClick={handleClose}
                >
                    {CloseIcon}
                </button>
            </div>
            <Text>{description}</Text>
        </div>
    );
};

export default Header;
