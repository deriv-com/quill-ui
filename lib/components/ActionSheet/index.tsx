import { ReactNode } from "react";
import ActionSheetRoot from "./root";
import ActionSheetHeader from "./header";
import ActionSheetContent from "./content";
import ActionSheetFooter from "./footer";
import ActionSheetHandleBar from "./handle-bar";
import ActionSheetTrigger from "./trigger";
import ActionSheetPortal from "./portal";

type ActionSheetType = {
    Root: typeof ActionSheetRoot;
    Header: typeof ActionSheetHeader;
    Content: typeof ActionSheetContent;
    Footer: typeof ActionSheetFooter;
    HandleBar: typeof ActionSheetHandleBar;
    Trigger: typeof ActionSheetTrigger;
    Portal: typeof ActionSheetPortal;
};

export const ActionSheet: ActionSheetType = ({
    children,
}: {
    children: ReactNode;
}) => {
    return <>{children}</>;
};

ActionSheet.Root = ActionSheetRoot;
ActionSheet.Header = ActionSheetHeader;
ActionSheet.Content = ActionSheetContent;
ActionSheet.Footer = ActionSheetFooter;
ActionSheet.HandleBar = ActionSheetHandleBar;
ActionSheet.Trigger = ActionSheetTrigger;
ActionSheet.Portal = ActionSheetPortal;

export default ActionSheet;
