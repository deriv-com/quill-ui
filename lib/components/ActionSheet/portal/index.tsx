import React from "react";
import {
    ComponentProps,
    forwardRef,
    useContext,
    useState,
    useEffect,
    useRef,
} from "react";
import { createPortal } from "react-dom";
import debounce from "lodash.debounce";
import HandleBar from "../handle-bar";
import "./portal.scss";
import { useSwipeBlock } from "@hooks/useSwipeBlock";
import { ActionSheetContext } from "../root";
import clsx from "clsx";

interface PortalProps extends ComponentProps<"div"> {
    shouldCloseOnDrag?: boolean;
    shouldDetectSwipingOnContainer?: boolean;
    showHandlebar?: boolean;
    fullHeightOnOpen?: boolean;
}

type TFooterChildProps = { handleClose: () => void };
type TFooterJSXConstructor = React.JSXElementConstructor<TFooterChildProps>;
type TFooterChild = React.ReactElement<
    TFooterChildProps,
    TFooterJSXConstructor
>;

const Portal = forwardRef<HTMLDivElement, PortalProps>(
    (
        {
            children,
            shouldCloseOnDrag,
            shouldDetectSwipingOnContainer = false,
            showHandlebar = true,
            fullHeightOnOpen = false,
            ...restProps
        },
        ref,
    ) => {
        const { show, handleClose, className, position, type, expandable } =
            useContext(ActionSheetContext);

        const [isVisible, setIsVisible] = useState(show);
        const animationTimerRef = useRef<ReturnType<typeof setTimeout>>();

        const toggleHandler = debounce(() => {
            setIsVisible(!isVisible);
            animationTimerRef.current = setTimeout(() => handleClose?.(), 500);
        }, 200);

        const childrenWithProps = React.Children.map(children, (child) => {
            if (
                React.isValidElement(child) &&
                (child?.type as TFooterJSXConstructor)?.name === "Footer"
            ) {
                return React.cloneElement(child as TFooterChild, {
                    handleClose: toggleHandler,
                });
            }
            return child;
        });

        const { height, containerRef, bindHandle, isScrolled, isLg } =
            useSwipeBlock({
                expandable,
                show,
                onClose: toggleHandler,
                shouldCloseOnDrag,
                fullHeightOnOpen,
            });

        useEffect(() => {
            setIsVisible(show);
            return () => clearTimeout(animationTimerRef.current);
        }, [show]);

        if (!show) return null;

        return (
            <>
                {createPortal(
                    <>
                        <div
                            className="quill-action-sheet--portal quill-action-sheet--portal--wrapper"
                            role="dialog"
                            data-state={isVisible ? "open" : "close"}
                            ref={ref}
                            {...restProps}
                        >
                            {type === "modal" && (
                                <div
                                    data-testid="dt-actionsheet-overlay"
                                    onClick={toggleHandler}
                                    className="quill-action-sheet--portal quill-action-sheet--portal__variant--modal"
                                />
                            )}
                            <div
                                className={clsx(
                                    "quill-action-sheet--root",
                                    `quill-action-sheet--root__show--${isVisible}`,
                                    `quill-action-sheet--root__position--${position}`,
                                    `quill-action-sheet--root__position--${position}__show--${isVisible}`,
                                    className,
                                )}
                                ref={containerRef}
                                style={{ height }}
                                {...(shouldDetectSwipingOnContainer &&
                                !isScrolled &&
                                !isLg &&
                                expandable
                                    ? bindHandle()
                                    : {})}
                            >
                                {showHandlebar && (
                                    <HandleBar
                                        {...(expandable || shouldCloseOnDrag
                                            ? bindHandle()
                                            : {})}
                                    />
                                )}

                                {childrenWithProps}
                            </div>
                        </div>
                    </>,
                    document.body,
                )}
            </>
        );
    },
);

export default Portal;
