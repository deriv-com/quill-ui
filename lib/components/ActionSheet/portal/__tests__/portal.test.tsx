import {
    screen,
    render as rtlRender,
    RenderResult,
    RenderOptions,
    act,
} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { RootPosition, RootProps } from "../../types";
import ActionSheet from "@components/ActionSheet";

const render = (
    ui: React.ReactElement,
    options?: RenderOptions & {
        wrapperProps?: RootProps;
    },
): RenderResult => {
    const { wrapperProps, ...renderOptions } = options || {};
    return rtlRender(ui, {
        wrapper: (props: RootProps) => (
            <ActionSheet.Root {...props} {...wrapperProps} />
        ),
        ...renderOptions,
    });
};

jest.mock("usehooks-ts", () => ({
    ...jest.requireActual("usehooks-ts"),
    useSsr: jest.fn().mockImplementation(() => ({
        isServer: false,
    })),
}));

describe("<ActionSheet.Portal/>", () => {
    it('should set the data-state attribute to "open" when the show is true', async () => {
        render(
            <>
                <ActionSheet.Trigger>Trigger</ActionSheet.Trigger>
                <ActionSheet.Portal>Portal</ActionSheet.Portal>
            </>,
        );
        const trigger = screen.getByText("Trigger");
        await act(async () => {
            await userEvent.click(trigger);
        });
        const state = screen.getByRole("dialog").getAttribute("data-state");
        expect(state).toBe("open");
    });

    it('should set the data-state attribute to "close" when the show is false', async () => {
        render(
            <>
                <ActionSheet.Trigger>Trigger</ActionSheet.Trigger>
                <ActionSheet.Portal>
                    <ActionSheet.Header closeIcon="Close" />
                </ActionSheet.Portal>
            </>,
        );
        const trigger = screen.getByText("Trigger");
        await act(async () => {
            await userEvent.click(trigger);
        });
        const close = screen.getByText("Close");
        await act(async () => {
            await userEvent.click(close);
        });
        const state = screen.getByRole("dialog").getAttribute("data-state");
        expect(state).toBe("close");
    });
    it("should render overlay when type prop is modal", () => {
        render(<ActionSheet.Portal>Portal</ActionSheet.Portal>, {
            wrapperProps: {
                type: "modal",
            },
        });
        const modalOverlay = screen.getByTestId("dt-actionsheet-overlay");
        expect(modalOverlay).toBeInTheDocument();
    });
    it("should not render overlay when type prop is non-modal", () => {
        render(<ActionSheet.Portal>Portal</ActionSheet.Portal>, {
            wrapperProps: {
                type: "non-modal",
            },
        });
        const modalOverlay = screen.queryByTestId("dt-actionsheet-overlay");
        expect(modalOverlay).not.toBeInTheDocument();
    });
    it("should close the action sheet when user clicked on overlay", async () => {
        render(
            <>
                <ActionSheet.Trigger>Trigger</ActionSheet.Trigger>
                <ActionSheet.Portal>Portal</ActionSheet.Portal>
            </>,
        );
        const trigger = screen.getByText("Trigger");
        await act(async () => {
            await userEvent.click(trigger);
        });
        const modalOverlay = screen.getByTestId("dt-actionsheet-overlay");
        await act(async () => {
            await userEvent.click(modalOverlay);
        });
        const state = screen.getByRole("dialog").getAttribute("data-state");
        expect(state).toBe("close");
    });
    it("should render handle bar when expandable prop is true", () => {
        render(<ActionSheet.Portal>Portal</ActionSheet.Portal>, {
            wrapperProps: { expandable: true },
        });
        const handleBar = screen.getByTestId("dt-actionsheet-handle-bar");
        expect(handleBar).toBeInTheDocument();
    });
    it("should not render handle bar when showHandlebar prop is false", () => {
        render(
            <ActionSheet.Portal showHandlebar={false}>
                Portal
            </ActionSheet.Portal>,
        );
        const handleBar = screen.queryByTestId("dt-actionsheet-handle-bar");
        expect(handleBar).not.toBeInTheDocument();
    });

    const positions: RootPosition[] = ["left", "right"];
    positions.forEach((position) => {
        it(`should render correctly with position ${position}`, () => {
            render(
                <ActionSheet.Portal>
                    <p>{position} portal</p>
                </ActionSheet.Portal>,
                {
                    wrapperProps: {
                        position,
                    },
                },
            );
            const actionRoot = screen.getByText(`${position} portal`);
            expect(actionRoot).toMatchSnapshot();
        });
    });
});
