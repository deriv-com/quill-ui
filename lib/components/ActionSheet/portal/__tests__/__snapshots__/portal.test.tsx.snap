// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionSheet.Portal/> should render correctly with position left 1`] = `
<p>
  left
   portal
</p>
`;

exports[`<ActionSheet.Portal/> should render correctly with position right 1`] = `
<p>
  right
   portal
</p>
`;
